src/Language/PureScript/TypeChecker/Entailment.hs

```diff
@@ -413,6 +414,28 @@ entails SolverOptions{..} constraint context hints =
       pure (arg1, arg2, TypeLevelString (mkString $ h' <> t'))
     consSymbol _ _ _ = Nothing
 
+    solveSymbolBreakOn :: [Type] -> Maybe [TypeClassDict]
+    solveSymbolBreakOn [arg0, arg1, arg2, arg3] = do
+      (arg0', arg1', arg2', arg3') <- breakOnSymbol arg0 arg1 arg2 arg3
+      let args' = [arg0', arg1', arg2', arg3']
+      pure [TypeClassDictionaryInScope [] 0 EmptyClassInstance [] C.SymbolBreakOn args' Nothing]
+    solveSymbolBreakOn _ = Nothing
+
+    -- breakOn by full, breaker, first, rest
+    breakOnSymbol :: Type -> Type -> Type -> Type -> Maybe (Type, Type, Type, Type)
+    breakOnSymbol breaker@(TypeLevelString breaker') full@(TypeLevelString full') _ _ = do
+      (first, rest) <- T.breakOn <$> decodeString breaker' <*> decodeString full'
+      pure (breaker, full, mkTLString first, mkTLString rest)
+      where mkTLString = TypeLevelString . mkString
+    breakOnSymbol breaker@(TypeLevelString breaker') _ first@(TypeLevelString first') rest@(TypeLevelString rest') = do
+      first'' <- decodeString first'
+      breaker'' <- decodeString breaker'
+      rest'' <- decodeString rest'
+      guard (T.isPrefixOf breaker'' rest'')
+      let full = first'' <> rest''
```

Comment:
Need to also guard on `first` not containing `breakOn` for the functional dependency to make sense.